### ISSUE TITLE
[Bug] [kdep] - kompiled directory path ignored by kdep

### ISSUE THREAD

**Issue by Baltoli** [runtimeverification/k#2194]
The path to the timestamp file generated by `kdep` does not respect the `-d` option and is therefore incorrect.

Steps to reproduce:
```shell
$ touch mod.k
$ mkdir -p some/nested/path
$ kdep -d some/nested/path mod.k
mod-kompiled/timestamp : \
    /Users/brucecollie/code/k/k-distribution/target/release/k/include/kframework/builtin/kast.md \
    /Users/brucecollie/code/k/k-distribution/target/release/k/include/kframework/builtin/domains.md \

```

The expected behaviour from kdep is to respect the `-d` option:
```shell
$ kdep -d some/nested/path mod.k
some/nested/path/mod-kompiled/timestamp : \
    /Users/brucecollie/code/k/k-distribution/target/release/k/include/kframework/builtin/kast.md \
    /Users/brucecollie/code/k/k-distribution/target/release/k/include/kframework/builtin/domains.md \

```

I've run `git bisect` and found that the bug was introduced in the big refactor of frontend options in [this commit](https://github.com/kframework/k/commit/309f81b73c0c3a447e78f4fd2bb413203e519a1d).

@SchmErik is there an obvious reason for this? I'm happy to investigate but you may know more quickly what's going on.

The workaround if this is breaking your Makefiles is just to edit the first line of the `.depend` file manually to point to the correct location.

**Comment by Baltoli** [comment]
It might also be worth implementing a regression test for this, similarly to the version flag tests we already have - it should just be a case of running `kdep -d ...` and checking that the first line points to the correct directory.

**Comment by SchmErik** [comment]
> The path to the timestamp file generated by `kdep` does not respect the `-d` option and is therefore incorrect.
> 
> Steps to reproduce:
> 
> ```shell
> $ touch mod.k
> $ mkdir -p some/nested/path
> $ kdep -d some/nested/path mod.k
> mod-kompiled/timestamp : \
>     /Users/brucecollie/code/k/k-distribution/target/release/k/include/kframework/builtin/kast.md \
>     /Users/brucecollie/code/k/k-distribution/target/release/k/include/kframework/builtin/domains.md \
> ```
> 
> The expected behaviour from kdep is to respect the `-d` option:
> 
> ```shell
> $ kdep -d some/nested/path mod.k
> some/nested/path/mod-kompiled/timestamp : \
>     /Users/brucecollie/code/k/k-distribution/target/release/k/include/kframework/builtin/kast.md \
>     /Users/brucecollie/code/k/k-distribution/target/release/k/include/kframework/builtin/domains.md \
> ```
> 
> I've run `git bisect` and found that the bug was introduced in the big refactor of frontend options in [this commit](https://github.com/kframework/k/commit/309f81b73c0c3a447e78f4fd2bb413203e519a1d).
> 
> @SchmErik is there an obvious reason for this? I'm happy to investigate but you may know more quickly what's going on.
> 

I think I know what's going on. Let me take a look and get back to you. We should add a regression test for this

> The workaround if this is breaking your Makefiles is just to edit the first line of the `.depend` file manually to point to the correct location.

**Comment by SchmErik** [comment]
I think I found the mistake. @Baltoli, I'm going to test this now:
```
diff --git a/kernel/src/main/java/org/kframework/kdep/KDepModule.java b/kernel/src/main/java/org/kframework/kdep/KDepModule.java
index cfed405b85..3b120de3cc 100644
--- a/kernel/src/main/java/org/kframework/kdep/KDepModule.java
+++ b/kernel/src/main/java/org/kframework/kdep/KDepModule.java
@@ -41,5 +41,5 @@ public class KDepModule extends AbstractModule {
     OuterParsingOptions outerParsingOptions(KDepOptions options) { return options.outerParsing; }
 
     @Provides
-    OutputDirectoryOptions outputDirectoryOptions(KompileOptions options) { return options.outputDirectory; }
+    OutputDirectoryOptions outputDirectoryOptions(KDepOptions options) { return options.outputDirectory; }
 }
```

**Comment by SchmErik** [comment]
I tested the above change and it now gives the expected result on a simple test case that I wrote

**Comment by Baltoli** [comment]
Fixed in #2195
